RHIDP-9010: AEM-related changes (#1426)
redhat-developer/red-hat-developers-documentation-rhdh: modules/customizing-the-appearance/proc-configuring-dynamic-plugin-menuitem.adoc

`icon`:: 
(Optional) Enter the icon name. You can use any of the following icons:
   ** Default icons, such as `home`, `group`, `category`, `extension`, and `school`. To use default icons, set the icon as an (`" "`) empty string.
   ** A custom icon, where _<my_icon>_ specifies the name of your custom icon
   ** An SVG icon, such as: `icon: <svg width="20px" height="20px" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg" fill="#ffffff">...</svg>`
   ** An HTML image, such as: `icon: https://img.icons8.com/ios-glyphs/20/FFFFFF/shop.png`

`title`:: 
(Optional) Enter the menu item title. Omit it when the title is already specified in the `dynamicRoutes` configuration under `menuItem.text`. To hide the title from the sidebar, set the title as an (`" "`) empty string.
// Update <4> for release 1.6 as this option (currently a workaround) would be added as a functionality. RHIDP-6333.

`priority`:: 

> (Optional) Sets the order in which menu items appear in the sidebar. The default priority is 0, which places the item at the bottom of the list. A higher priority value places the item higher in the sidebar. You can define this field for each section.

Reviewer: ```suggestion
(Optional) Enter an integer value to set the order in which menu items appear in the sidebar.
The default priority is 0, which places the item at the bottom of the list.
A higher priority value places the item higher in the sidebar.
You can define this field for each section.
```

`parent`:: 
(Optional) Enter the parent menu item under which the current item is nested. If this field is used, the parent menu item must be defined elsewhere in the `menuItems` configuration of any enabled plugin. You can define this field for each section.

`enabled`:: 

> (Optional) If this field is used to hide the menu item from the sidebar, set the value to `false`. To display the menu item in the sidebar, set the value to `true`.

Reviewer: ```suggestion
(Optional) Enter `false` to hide the menu item from the sidebar. 
Enter `true` to display the menu item in the sidebar.
```

            icon: fooIcon
            text: Foo Plugin Page
      menuItems:
        my-plugin:
          priority: 10
          parent: favorites
        favorites:
          icon: favorite
          title: Favorites
          priority: 100
----
`my-plugin`:: 
Matches the value of the `path` field in `dynamicRoutes`.

`priority`:: 

> Controls order of plugins under the parent menu item.

Reviewer: ```suggestion
Enter an integer value to set the order in which plugins appear in the parent menu item.
Enter `0` to place the item at the bottom of the list.
Enter a higher priority value, such as `10` or `100` to place the item higher in the list.
```

`parent`:: 

> Nests this plugin under the `favorites` parent menu item.

Reviewer: ```suggestion
Enter the parent menu item id to nest this plugin under, such as `favorites`.
```

`favorites`:: 
Configuration for the parent menu item.

> 
> `icon`:: 
> Displays the `favorite` icon from the {product-very-short} system icons.

Reviewer: ```suggestion
```

Repeated content

`title`:: 
Displays the title name for the parent menu item.

> 
> `priority`:: 
> Order of the `favourites` menu item in the sidebar.

Reviewer: ```suggestion
```

Repeated content
